Server/middleware/auth.middleware.js

```diff
@@ -0,0 +1,45 @@
+import jwt from "jsonwebtoken";
+import Joi from "joi";
+import "dotenv/config";
+import User from "../models/User.model.js";
+import Responses from "../helpers/responses";
+
+class Auth {
+   static signin = async (req, res, next) => {
+    const schema = Joi.object({
+      email: Joi.string().trim().required().min(6),
+      password: Joi.string().trim().required().min(8),
+      role: Joi.string().trim().required().min(4),
```

Comment:
@elvisiraguha it would be better to have the role either in the database or embedded in the token. The  API users don't really need to provide their role.